[Bug]: `storybook init` fails in React Native when the project already has some dependencies

### Describe the bug

Storybook init for React Native adds a combination of dependencies, which seem to be the renderer + addons + the peer dependencies of the addons:
```
'react-native-safe-area-context',
'@react-native-async-storage/async-storage',
'@react-native-community/datetimepicker',
'@react-native-community/slider',
'@storybook/addon-ondevice-controls',
'@storybook/addon-ondevice-actions',
'@storybook/react-native',
```

However storybook init tries to add them at [a particular version](https://github.com/storybookjs/storybook/blob/3bf95ad465d14c91a1a1040462a74389a4cd065f/code/lib/cli/src/generators/REACT_NATIVE/index.ts#L30) and that causes YARN2 error as they are already installed, thus storybook init fails.

These were the deps in my project which caused the failure:
```
"dependencies": {
    "@expo/webpack-config": "^19.0.0",
    "@react-native-async-storage/async-storage": "1.18.2",
    "@react-native-picker/picker": "2.4.10",
    "@react-navigation/bottom-tabs": "^6.5.11",
    "@react-navigation/native": "^6.1.9",
    "@reduxjs/toolkit": "^1.9.7",
    "@types/react": "~18.2.14",
    "@types/react-native": "~0.72.6",
    "expo": "^49.0.18",
    "expo-av": "~13.4.1",
    "expo-linear-gradient": "~12.3.0",
    "expo-status-bar": "~1.6.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.72.6",
    "react-native-calendars": "^1.1302.0",
    "react-native-linear-gradient": "^2.8.3",
    "react-native-picker-select": "^9.0.0",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-web": "~0.19.6",
    "react-redux": "^8.1.3",
    "redux-persist": "^6.0.0",
    "typescript": "^5.1.3"
  },
```

### To Reproduce

_No response_

### System

_No response_

### Additional context

_No response_

@yannbf btw for npm I get an install error since that version of async storage has a peer dep that doesn't match your react native version

```
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@"^0.0.0-0 || 0.60 - 0.72 || 1000.0.0" from @react-native-async-storage/async-storage@1.18.2
npm ERR! node_modules/@react-native-async-storage/async-storage
npm ERR!   @react-native-async-storage/async-storage@"1.18.2" from the root project
```

--- a/code/lib/cli/src/generators/REACT_NATIVE/index.ts
+++ b/code/lib/cli/src/generators/REACT_NATIVE/index.ts
@@ -14,12 +14,15 @@ const generator = async (
 
   const reactVersion = packageJson.dependencies.react;
 
-  const packagesToResolve = [
-    // addon-ondevice-controls peer deps
+  const controlsPeerDependencies = [
     'react-native-safe-area-context',
     '@react-native-async-storage/async-storage',
     '@react-native-community/datetimepicker',
     '@react-native-community/slider',
+  ].filter((dep) => !packageJson.dependencies[dep] && !packageJson.devDependencies[dep]);
+
+  const packagesToResolve = [
+    ...controlsPeerDependencies,
     '@storybook/addon-ondevice-controls',
     '@storybook/addon-ondevice-actions',
     '@storybook/react-native',

